### PR TITLE
plutus-tx: Quoted 'Int' operators

### PR DESCRIPTION
**Pull request by j-mueller**
* Export the operators from 'Num Int' and 'Ord Int' as quoted template
  Haskell expressions for greater consistency with the rest of the
  Plutus prelude

### CODE REVIEW

File: plutus-tx/src/Language/PlutusTx/Prelude/Stage0.hs [hunk 1]
```diff
@@ -64,6 +65,86 @@ or = [|| \(l :: Bool) (r :: Bool) -> if l then True else r ||]
 not :: Q (TExp (Bool -> Bool))
 not = [|| \(a :: Bool) -> if a then False else True  ||]
 
+-- | Greater than 
+--
+--   >>> $$([|| $$(gt) 2 1 ||])
+--   True
+--
+gt :: Q (TExp (Int -> Int -> Bool))
+gt = [|| (>) ||]
```
Comment (michaelpj) [inline plutus-tx/src/Language/PlutusTx/Prelude/Stage0.hs]: You should be able to use the non-symbolic ones in `Builtins`. That way I can delete the hacky typeclass handling entirely ;)

Comment (j-mueller) [reply plutus-tx/src/Language/PlutusTx/Prelude/Stage0.hs]: I'll look into that!

**Comment by j-mueller** [comment]
@michaelpj I uses the builtins everywhere, but I also replaced `mustBeReplaced` in `Language.PlutusTx.Builtins` with the Haskell definitions where possible. That way it is more likely that a piece of TH code works on-chain and off-chain.

The plugin identifies builtins by their qualified name, so it shouldn't be affected, correct? (all the tests still pass)

**Comment by michaelpj** [comment]
No, please don't do that! The builtins are *only* valid in on-chain code, and it's very deliberate that they blow up in any other context.

**Comment by j-mueller** [comment]
That makes it much more difficult to write some "contract logic" that is used by the wallet and the on-chain code alike. 

For example, the swap contract has a margin calculation that I expressed as a `Q (TExp (Swap -> Rational -> Value))` which is used in several places. And it only works because it currently does the right thing when you evaluate it in Haskell. 

If we make it impossible to evaluate functions from `Language.PlutusTx.Builtins` in Haskell then our code really isn't portable anymore and we have to duplicate a lot of the internals of our contracts.

**Comment by j-mueller** [comment]
An immediate conequence is that we have to delete the doctests for everything that use the builtins (or change them to compile to PLC and show the PLC result)

**Comment by j-mueller** [comment]
build is failing because I removed the "buildable: False" line from the wallet-api doctests, I'll put up a different PR shortly that deals with this

**Comment by michaelpj** [comment]
You could potentially nick what I did for the `plutus-tx` ones, which is quite nice and uses `cabal doctest`.

**Comment by j-mueller** [comment]
> You could potentially nick what I did for the plutus-tx ones, which is quite nice and uses cabal doctest.

That's what I ended up doing, it's a lot nicer than the current file.

**Comment by michaelpj** [comment]
> I think, we really want to be able to use the same code on-chain and off-chain. We go to all the trouble to use the same language (or at least have a subset relationship). It would be pity to not be able to make use of that fact by sharing code.

I agree with this, the thing I'm torn about is whether we should achieve this by:
- Having people use the "normal" operations, and intercepting them to turn them into the Plutus ones (what we do now).
- Having people use the "Plutus" operations, and having them have normal Haskell definitions too so they work off-chain.

I think I've come down on the latter, for the following reason: one thing we want to do is have proper sized integers. These will likely have operations that aren't existing Haskell operations, and people may want to manipulate them both on and off-chain. So I don't see any way to do this nicely other than having the Plutus operations work off-chain by having Haskell implementations.

So I'm :+1: on this PR now. Once it's in I'll go in and add a note about the design here.

**Pull request merged**